src/pages/NewPostPage.tsx

```diff
@@ -0,0 +1,164 @@
+import { Button, Box, CircularProgress, Input, TextField } from "@mui/material";
+import { useState } from "react";
+import { CKEditor } from "@ckeditor/ckeditor5-react";
+import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
+import { AddNewPost } from "../redux/NewPostSlice";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatchType, RootState } from "../redux/Store";
+import CustomModal from "../components/CustomModal";
+import { useNavigate } from "react-router-dom";
+import { resetSuccess } from "../redux/NavBarSlice";
+interface newPostDataType {
+  title: string;
+  picture: File | null;
+  description: string;
+}
+export const NewPostPage = () => {
+  const [openCloseModal, setOpenCloseModal] = useState(true);
+  const [inputFieldError, setInputFieldError] = useState({
+    title: false,
+    description: false,
+  });
+  const navigate = useNavigate();
+  const dispatch = useDispatch<AppDispatchType>();
+  const newPostStatus = useSelector((state: RootState) => state.newPostDetail);
+
+  const LoggedInUserData = useSelector((state: RootState) => state.navbarData);
+
+  const [newPostData, setNewPostData] = useState<newPostDataType>({
+    title: "",
+    picture: null,
+    description: "",
+  });
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (!(event.target.value.trim() === "")) {
+      const postData = {
+        ...newPostData,
+        [event.target.id]: event.target.value,
+      };
+      setNewPostData(postData);
+    }
+  };
+  const handleSubmitNewPost = () => {
+    if (newPostData.picture) {
+      dispatch(
+        AddNewPost({
+          title: newPostData.title,
+          photo: newPostData.picture,
+          description: newPostData.description,
+          firstName: LoggedInUserData.userDetails!.firstName,
+          lastName: LoggedInUserData.userDetails!.lastName,
+          updatedBy: LoggedInUserData.userDetails!.uid,
+          profilePhotoPath: LoggedInUserData.userDetails!.profilePhotoPath,
+        })
+      );
+    }
+  };
+  const handleSuccessUpdateInfo = async () => {
+    navigate("/userProfile");
+    dispatch(resetSuccess());
+  };
+  const handleOnBlur = (event: any) => {
+    if (newPostData.title.trim() === "") {
+      setInputFieldError({
+        ...inputFieldError,
+        title: true,
+      });
+    } else {
+      setInputFieldError({
+        ...inputFieldError,
+        title: false,
+      });
+    }
+  };
+  const handleOnBlurDescription = () => {
+    if (newPostData.description.trim() === "") {
+      setInputFieldError({
+        ...inputFieldError,
+        description: true,
+      });
+    } else {
+      setInputFieldError({
+        ...inputFieldError,
+        description: false,
+      });
+    }
+  };
```

Comment:
improve function handleOnBlur and handleOnBlurDescription, it can be done sing a single function as well as @Rishab-innovative